trin-core/src/portalnet/events.rs

```diff
@@ -110,20 +112,49 @@ impl PortalnetEvents {
     // This could be handled in the UtpListener impl, but for consistency with data handling
     // and clarity it can be put here
     async fn process_utp_byte_stream(&mut self) {
-        for (_, conn) in self.utp_listener.utp_connections.iter_mut() {
+        for (_, conn) in self
+            .utp_listener
+            .write_with_warn()
+            .await
+            .utp_connections
+            .iter_mut()
+        {
             let received_stream = conn.recv_data_stream.clone();
-            if received_stream.is_empty() {
-                continue;
-            }
 
-            let message_len = u32::from_be_bytes(received_stream[0..4].try_into().unwrap());
+            match UtpMessage::decode(&received_stream[..]) {
+                Ok(message) => {
+                    match self
+                        .utp_listener
+                        .read_with_warn()
+                        .await
+                        .listening
+                        .get(&conn.conn_id_recv)
+                    {
+                        Some(message_type) => match message_type {
+                            UtpMessageId::OfferAcceptStream => {
+                                let payload = UtpAccept::from_ssz_bytes(&message.payload[..])
+                                    .unwrap()
+                                    .message;
 
-            if message_len + 4 >= received_stream.len() as u32 {
-                // handle data function(&received_stream[4..(message_len + 4)]
+                                for (key, content) in payload {
+                                    store(key, content).unwrap();
+                                }
+                            }
+                        },
+                        _ => warn!("uTP listening HashMap doesn't have uTP stream message type"),
+                    }
 
-                // Removed the handled data from the buffer
-                conn.recv_data_stream = received_stream[((message_len + 4) as usize)..].to_owned();
+                    // message was successfully handled, remove it from the buffer
+                    conn.recv_data_stream = received_stream[(message.len() as usize)..].to_owned();
+                }
+                // Ignore messages which are too short
+                _ => {}
```

Comment:
Maybe a `debug!`, while this code is new? We'll probably want to remove it eventually, once we see that it's not popping up constantly.